src/main/java/com/firebolt/jdbc/service/FireboltEngineInformationSchemaService.java

```diff
@@ -11,29 +11,44 @@
 import java.sql.PreparedStatement;
 import java.sql.ResultSet;
 import java.sql.SQLException;
+import java.util.Collection;
+import java.util.TreeSet;
+import java.util.stream.Stream;
 
+import static java.lang.String.CASE_INSENSITIVE_ORDER;
 import static java.lang.String.format;
+import static java.util.stream.Collectors.toCollection;
 
 @RequiredArgsConstructor
 @CustomLog
 public class FireboltEngineInformationSchemaService implements FireboltEngineService {
     private static final String ENGINE_URL = "url";
     private static final String STATUS_FIELD = "status";
     private static final String ENGINE_NAME_FIELD = "engine_name";
-    private static final String RUNNING_STATUS = "running";
+    private static final Collection<String> RUNNING_STATUSES = Stream.of("running", "ENGINE_STATE_RUNNING").collect(toCollection(() -> new TreeSet<>(CASE_INSENSITIVE_ORDER)));
     private static final String ENGINE_QUERY =
             "SELECT engs.url, engs.attached_to, dbs.database_name, engs.status, engs.engine_name " +
```

Comment:
In v2, engs.attached_to will always be empty. We could consider just removing it from this query + our check if it is empty or not - see https://firebolt-analytics.slack.com/archives/C02QFG24RN1/p1704838365433459?thread_ts=1702678581.390369&cid=C02QFG24RN1 for some details.

Reply:
The check below 
```
                String attachedDatabase = rs.getString("attached_to");
                if (attachedDatabase == null) {
                    throw new FireboltException(format("The engine with the name %s is not attached to any database", engine));
                }
```

will still fail for any v2 engine.

Reply:
Fixed. Now we have 3 implementations of `FireboltEngineService`; one of them `FireboltEngineVersion2Service` implements this logic